src/Acme.UI/Infrastructure/Handlers/MemberSaveHandler.cs

```diff
@@ -0,0 +1,193 @@
+﻿using Umbraco.Core.Composing;
+using Umbraco.Core;
+using Umbraco.Core.Services;
+using Umbraco.Core.Services.Implement;
+using Umbraco.Core.Events;
+using Umbraco.Core.Models;
+using Examine;
+using Umbraco.Examine;
+using System.Linq;
+using System;
+using System.Collections.Generic;
+using Acme.UI.Helper.Extensions;
+using Published = Umbraco.Web.PublishedModels;
+using System.Linq.Expressions;
+using Umbraco.Web;
+
+namespace Acme.UI.Infrastructure.Handlers
+{
+    public class MemberSaveHandlerComposer : IUserComposer
+    {
+        public void Compose(Composition composition)
+        {
+            composition.Components().Append<MemberSaveHandler>();
+        }
+    }
+
+    public class MemberSaveHandler : IComponent
+    {
+        private Published.Member _member;
+        private Published.Physiotherapist _physio;
+        private IUmbracoContextFactory _contextFactory;
+        private IContentService _contentService;
+        
+        public MemberSaveHandler(IContentService contentService, IUmbracoContextFactory contextFactory)
+        {
+            _member = new Published.Member(null);
+            _physio = new Published.Physiotherapist(null);
+            _contentService = contentService;
+            _contextFactory = contextFactory;
+        }
+
+        public void Initialize()
+        {
+            MemberService.Saving += OnMemberSavingHandler;
+        }
+
+        /// <summary>
+        /// It is responsible to create newly activated physio's data folder and link it to his/her membership record
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="args"></param>
+        private void OnMemberSavingHandler(IMemberService sender, SaveEventArgs<IMember> args)
+        {
+            List<string> processedNodesIds = new List<string>();
+
+            foreach (IMember member in args.SavedEntities)
+            {
+                if (processedNodesIds.FirstOrDefault(id => id == member.Id.ToString()) == null)
```

Comment:
There are 4 levels of nested if's that become hard to track and understand, I would suggest we attempt to refactor these into functions, each describing what its scenario deals with.
Lets try do that on our next round of changes in this file.